File: modules/annotations/src/main/java/symphony/annotations/java/GQLInterface.java
package symphony.annotations.java;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/** Annotation to make an interface type. */
@Documented
@Retention(RetentionPolicy.CLASS)
@Target({ElementType.TYPE})
public @interface GQLInterface {
  /**
   * @return Optionally provide a list of field names that should be excluded from the interface.
   */
  String[] excludedFields() default {};
}

File: modules/annotations/src/main/scala/symphony/annotations/scala/GQLInterface.scala
package symphony.annotations.scala

import scala.annotation.StaticAnnotation

/**
 * Annotation to make a sealed trait an interface instead of a union type or an enum.
 *
 * @param excludedFields Optionally provide a list of field names that should be excluded from the interface.
 */
case class GQLInterface(excludedFields: String*) extends StaticAnnotation
